pkg/migrations/op_set_unique.go

```diff
@@ -25,12 +26,43 @@ func (o *OpSetUnique) Start(ctx context.Context, conn db.DB, latestSchema string
 	table := s.GetTable(o.Table)
 	column := table.GetColumn(o.Column)
 
-	// Add a unique index to the new column
-	if err := addUniqueIndex(ctx, conn, table.Name, column.Name, o.Name); err != nil {
-		return nil, fmt.Errorf("failed to add unique index: %w", err)
+	for retryCount := 5; retryCount > 0; retryCount-- {
+		// Add a unique index to the new column
+		if err := addUniqueIndex(ctx, conn, table.Name, column.Name, o.Name); err != nil {
+			return nil, fmt.Errorf("failed to add unique index: %w", err)
+		}
+
+		// Make sure Postgres is done creating the index
+		isInProgress := true
+		for isInProgress {
+			var err error
+			isInProgress, err = isIndexInProgress(ctx, conn, s.Name, o.Name)
+			if err != nil {
+				return nil, err
+			}
+			// Still in progress, sleep for 0.5 seconds and check again
+			time.Sleep(500 * time.Millisecond)
+		}
```

Comment:
This is always going to sleep for 500ms, regardless of whether the index creation succeeds on the first iteration. Maybe we can rewrite to avoid the sleep if `isInProgress` is false?